### ISSUE TITLE
Linechart does not work: dataByDate[0] is undefined

### ISSUE THREAD

**Issue by ervinfetic** [eventbrite/britecharts#545]
The error happens only in Firefox but when I try with Chrome it works well. I am pretty sure when I tried this example few days ago even with Firefox it worked.

I tried the same example on your official page I got the same error. I will put screenshots below.

I am using following versions of libs:
1. d3 v4.13.0
2. britecharts v2.6.0 but I tried also v3.0.0

I imported libs via these urls:
https://cdnjs.cloudflare.com/ajax/libs/d3/4.13.0/d3.min.js
https://cdn.jsdelivr.net/npm/britecharts@2.6.0/dist/css/britecharts.min.css
https://cdn.jsdelivr.net/npm/britecharts@2.6.0/dist/bundled/britecharts.min.js

## Screenshots (if appropriate):
<img width="1185" alt="firefox_1" src="https://user-images.githubusercontent.com/15788855/37083318-7686b73c-21ef-11e8-963c-341e981c64c3.png">

![firefox_2](https://user-images.githubusercontent.com/15788855/37083389-a237eb8a-21ef-11e8-8b7a-7c902918806b.png)

**Comment by Golodhros** [comment]
Hi @ervinfetic, thanks for opening this issue!

Lately I found an error similar to yours. In my case was related to the date format, as it seems that different browsers support different date formats when we do `new Date(DATE)`. Could you check this? It would be as easy as to grab the date you are using and, in the Firefox console, type `new Date(DATE)` and see if the response is something valid.

Let us know what comes up from this!

**Comment by ervinfetic** [comment]
Hi @Golodhros,

Thanks for the response.

I tried what you said. The problem was date in different browsers. In your examples on [https://codepen.io/Britecharts](url) for lineCharts you used the following date format "27-Jun-15". 

When you try to do this `new Date("27-Jun-15");` in Firefox you will get an error `("Invalid Date")` while in Chrome it works well.

So I changed all my dates into format "YYYY-MM-DD" and it work now well.
